### ISSUE TITLE
ppp device stuck on configuring DHCP6 with IPv6AcceptRA=no and UseAddress=no and UplinkInterface=:self

### ISSUE THREAD

**Issue by pimzand** [systemd/systemd#29979]
### systemd version the issue has been seen with

254

### Used distribution

Fedora 39

### Linux kernel version used

6.5.9-300.fc39.x86_64

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

A PPPoE connection as described in #26681 would reach the "configured" state, like it did in Fedora 38 with systemd 253.

### Unexpected behaviour you saw

The PPPoE connection is stuck in "configuring" state.

### Steps to reproduce the problem

Set up a PPPoE connection using systemd-networkd and pppd as described in #26681, and include the (then) solution provided in https://github.com/systemd/systemd/issues/26681#issuecomment-1556597979.

A complete description for the ppp network would be:
```
[Match]
Type = ppp

[Network]
DHCP = ipv6
LLMNR = no
IPv6AcceptRA = no
KeepConfiguration = yes
DHCPPrefixDelegation = yes

[Link]
RequiredForOnline = yes

[DHCPv6]
UseDelegatedPrefix = yes
UseAddress = no
UseDNS = no

[DHCPPrefixDelegation]
UplinkInterface = :self
Assign = no
```

This setup is expecting a ppp connection that will be configured using IPCP with a public IPv4 address and a link-local IPv6 address. DHCPv6 would be used only to configure addressing and routing on other interfaces using prefix delegation.

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCPv6 client: Timeout T1
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCPv6 client: State changed: bound -> renew
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCPv6 client: Sent Renew
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCPv6 client: Next retransmission in 9s
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCPv6 client: Processed Reply message
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCPv6 client: T1 expires in 21h 52min 41s
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCPv6 client: T2 expires in 1d 11h 30min 52s
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCPv6 client: Valid lifetime expires in 3d
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCPv6 client: State changed: renew -> bound
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCP: received delegated prefix aaaa:bbbb:cccc::/48
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: Updated expiration timer for DHCPv6 route (configuring,configured): dst: aaaa:bbbb:cccc::/48, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254), proto: dhcp, type: unreachable, nexthop: 0, priority: 1024, flags: n/a
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: Requesting DHCPv6 route (configuring,configured): dst: aaaa:bbbb:cccc::/48, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254), proto: dhcp, type: unreachable, nexthop: 0, priority: 1024, flags: n/a
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: Requesting DHCP-PD route (configuring,configured): dst: aaaa:bbbb:cccc::/64, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254), proto: dhcp, type: unicast, nexthop: 0, priority: 256, flags: n/a
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: Assigned prefix aaaa:bbbb:cccc::/64
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: link_check_ready(): DHCPv6 IA_PD prefix is assigned, but DHCPv6 protocol is not finished yet.
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: Setting DHCPv6 addresses and routes
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: link_check_ready(): DHCPv6 IA_PD prefix is assigned, but DHCPv6 protocol is not finished yet.
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: Configuring DHCPv6 route (requesting,configuring,configured): dst: aaaa:bbbb:cccc::/48, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254), proto: dhcp, type: unreachable, nexthop: 0, priority: 1024, flags: n/a
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: Configuring DHCP-PD route (requesting,configuring,configured): dst: aaaa:bbbb:cccc::/64, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254), proto: dhcp, type: unicast, nexthop: 0, priority: 256, flags: n/a
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: dhcp6_check_ready(): no DHCPv6 address is ready.
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: DHCP-PD addresses and routes set.
Nov 09 21:50:27 fedora systemd-networkd[1766]: ppp0: link_check_ready(): DHCPv6 IA_PD prefix is assigned, but DHCPv6 protocol is not finished yet.
```

**Comment by yuwata** [comment]
Thank you!

**Comment by yuwata** [comment]
Fix is waiting in #29981.

**Comment by pimzand** [comment]
confirmed fixed in Fedora systemd-254.7-1.fc39
